int64 leftRange
int64 rightRange
